lib/dcell/registries/cassandra_adapter.rb

```diff
@@ -44,83 +47,40 @@ def initialize(options)
 
         cass = Cassandra.new(keyspace, options[:servers])
 
-        @node_registry   = NodeRegistry.new(cass, columnfamily)
-        @global_registry = GlobalRegistry.new(cass, columnfamily)
-      end
-
-      def remove_node(node)
-        @node_registry.remove node
-      end
-
-      def clear_all_nodes
-        @node_registry.clear_all
-      end
-
-      def clear_globals
-        @global_registry.clear_all
+        @node_registry = Registry.new(cass, 'nodes', columnfamily)
+        @global_registry = Registry.new(cass, 'globals', columnfamily)
       end
 
-      class NodeRegistry
-        def initialize(cass, cf)
-          @cass = cass
-          @cf = cf
-        end
-
-        def get(node_id)
-          @cass.get @cf, "nodes", node_id
-        end
-
-        def set(node_id, addr)
-          @cass.insert @cf, "nodes", { node_id => addr }
-        end
-
-        def nodes
-          @cass.get(@cf, "nodes").keys
-        end
-
-        def remove(node_id)
-          @cass.remove @cf, "nodes", node_id
-        end
-
-        def clear_all
-          @cass.remove @cf, "nodes"
-        end
-      end
-
-      def get_node(node_id);       @node_registry.get(node_id) end
-      def set_node(node_id, addr); @node_registry.set(node_id, addr) end
-      def nodes;                   @node_registry.nodes end
-
-      class GlobalRegistry
-        def initialize(cass, cf)
+      class Registry
+        def initialize(cass, table, cf)
           @cass = cass
+          @table = table
           @cf = cf
         end
 
         def get(key)
-          string = @cass.get @cf, "globals", key.to_s
-          Marshal.load string if string
+          value = @cass.get @cf, @table, key.to_s
+          return nil unless value
+          MessagePack.unpack(value, options={:symbolize_keys => true})
         end
 
-        # Set a global value
         def set(key, value)
-          string = Marshal.dump value
-          @cass.insert @cf, "globals", { key.to_s => string }
+          id = Utils::uuid
```

Comment:
Okay, I see, this might be a bit of copypasta from a previous PR that used this odd style.